Content.Shared/_Impstation/Tools/SolutionRefuelSystem.cs

```diff
@@ -0,0 +1,96 @@
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Chemistry.Components.SolutionManager;
+using Content.Shared.Database;
+using Content.Shared.DoAfter;
+using Content.Shared.Examine;
+using Content.Shared.IdentityManagement;
+using Content.Shared.FixedPoint;
+using Content.Shared.Interaction;
+using Content.Shared.Tools.Components;
+using Content.Shared.Whitelist;
+namespace Content.Shared.Tools.Systems;
+
+public abstract partial class SharedToolSystem
+{
+    public void InitializeSolutionRefuel()
+    {
+        SubscribeLocalEvent<SolutionRefuelComponent, ExaminedEvent>(SolutionRefuelExamine);
+        SubscribeLocalEvent<SolutionRefuelComponent, AfterInteractEvent>(OnSolutionRefuelAfterInteract);
+    }
+
+    private string GetName(Entity<SolutionRefuelComponent> entity)
+    {
+        if (entity.Comp.Name == null)
+            return Identity.Name(entity, EntityManager);
+        return Loc.GetString(entity.Comp.Name);
+    }
+
+    public (FixedPoint2 fuel, FixedPoint2 capacity) GetSolutionFuelAndCapacity(EntityUid uid, SolutionRefuelComponent? welder = null, SolutionContainerManagerComponent? solutionContainer = null)
+    {
+        if (!Resolve(uid, ref welder, ref solutionContainer))
+            return default;
+
+        if (!SolutionContainerSystem.TryGetSolution(
+                (uid, solutionContainer),
+                welder.FuelSolutionName,
+                out _,
+                out var fuelSolution))
+        {
+            return default;
+        }
+
+        return (fuelSolution.GetTotalPrototypeQuantity(welder.FuelReagent), fuelSolution.MaxVolume);
+    }
+
+    private void SolutionRefuelExamine(Entity<SolutionRefuelComponent> entity, ref ExaminedEvent args)
+    {
+        using (args.PushGroup(nameof(SolutionRefuelComponent)))
+        {
+            if (args.IsInDetailsRange)
+            {
+                var (fuel, capacity) = GetSolutionFuelAndCapacity(entity.Owner, entity.Comp);
```

Comment:
```suggestion
               if (!TryGetSolutionFuelAndCapacity(ent, out var (YOUR OUT))
               push an loc string for the error
```
refer to code in `SharedSolutionContainerSystem`